src/common/datetime/format_date_time.ts

```diff
@@ -11,21 +11,36 @@ export const formatDateTime = (
   dateObj: Date,
   locale: FrontendLocaleData,
   config: HassConfig
-) => formatDateTimeMem(locale, config.time_zone).format(dateObj);
+) => formatDateTimeMem(locale, config?.time_zone).format(dateObj);
 
 const formatDateTimeMem = memoizeOne(
   (locale: FrontendLocaleData, serverTimeZone: string) =>
     new Intl.DateTimeFormat(locale.language, {
       year: "numeric",
       month: "long",
       day: "numeric",
-      hour: useAmPm(locale) ? "numeric" : "2-digit",
+      hour: locale && useAmPm(locale) ? "numeric" : "2-digit",
       minute: "2-digit",
-      hourCycle: useAmPm(locale) ? "h12" : "h23",
+      hourCycle: locale && useAmPm(locale) ? "h12" : "h23",
       timeZone: resolveTimeZone(locale.time_zone, serverTimeZone),
     })
 );
 
+export const formatDateTimeWithBrowserDefaults = (dateObj: Date) =>
+  formatDateTimeWithBrowserDefaultsMem().format(dateObj);
+
+const formatDateTimeWithBrowserDefaultsMem = memoizeOne(
+  () =>
+    new Intl.DateTimeFormat(undefined, {
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+      hour: "2-digit",
+      minute: "2-digit",
+      hourCycle: "h23",
```

Comment:
Why would we add this? This means we will always display it in 24 hour format, also for browsers that have `en-US` for example that should use 12 hour format?